pkg/server/ca/manager/journal.go

```diff
@@ -196,30 +206,122 @@ func (j *Journal) UpdateJWTKeyStatus(issuedAt time.Time, status journal.Status)
 		return fmt.Errorf("no journal entry found issued at: %v", issuedAtUnix)
 	}
 
-	if err := j.save(); err != nil {
+	if err := j.save(ctx); err != nil {
 		j.entries.JwtKeys = backup
 		return err
 	}
 
 	return nil
 }
 
-func (j *Journal) save() error {
-	return saveJournalEntries(j.path, j.entries)
+func (j *Journal) setEntries(entries *journal.Entries) {
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	j.entries = entries
 }
 
-func saveJournalEntries(path string, entries *JournalEntries) error {
-	entriesBytes, err := proto.Marshal(entries)
+// saveInDatastore saves the provided marshaled entries in the datastore.
+// If caJournalID has not been defined yet (it's value is 0), it first finds
+// the CA journal records that corresponds to this server. In case that there is
+// no CA record for this server, it creates one.
+// The ID of the CA journal record that was saved is returned, in addition to
+// the error (if any) of the operation.
+func (j *Journal) saveInDatastore(ctx context.Context, entriesBytes []byte) (caJournalID uint, err error) {
+	// Check if we already identified what's the CA journal for this server in
+	// the datastore.
+	if j.caJournalID == 0 {
+		// We haven't identified what's the CA journal for this server.
+		// Look for a record that has an active X509 authority ID that matches
+		// one of our key manager keys.
+		caJournal, err := j.findCAJournal(ctx)
+		if err != nil {
+			return 0, fmt.Errorf("failed to get CA journal from database: %w", err)
+		}
+		if caJournal == nil {
```

Comment:
I think that we can safely remove this call to findCAJournal since a save operation will always happen after the loadJournalFromDS operation, which is the one that calls to findCAJournal and will update j.caJournalID.